server/src/main/java/org/opensearch/indices/ShardLimitValidator.java

```diff
@@ -64,12 +64,23 @@ public class ShardLimitValidator {
         Setting.Property.Dynamic,
         Setting.Property.NodeScope
     );
+
+    public static final Setting<Boolean> SETTING_CLUSTER_IGNORE_HIDDEN_INDEXES = Setting.boolSetting(
+        "cluster.ignore_hidden_indexes",
+        true,
```

Comment:
This being `true` by default is backwards incompatible. I think you want it to be `false` so you can backport this change into 2.x. If we want to flip it to true in 3.0 we can do this with documentation.

Reply:
I do agree @dblock that making this property default to be true might cause some in-compatibility issues. I will make this to be false as default for now and in later version we can change the default value for this.

Reply:
@anasalkouz will make this property false to avoid that issue